.github/workflows/ci-smart-contract.yml

```diff
@@ -0,0 +1,180 @@
+        with:
+          repo: 'Finschia/cosmwasm'
+          version: tags/${{ matrix.versions }}
+          regex: true
+          file: '.*'
+          target: 'contracts/'
+      - name: Download build
+        uses: actions/download-artifact@v3
+        with:
+          name: fnsad
+          path: ./build
+      - name: Build install
+        run: make install
+      - name: Set configure file
+        run: ./init_single.sh
+        shell: bash
+      # sleep 1 second to wait for finschia to start up
+      - name: Start local node
+        id: start
+        run: |
+          fnsad start &
+          sleep 1
+      - name: CI test
+        run: bash ./ci_test/queue_ci.sh
+        shell: bash
+      - name: Stop local node
+        if: ${{ always() && steps.start.conclusion == 'success' }}
+        run: kill $(ps -ef | grep "fnsad start" | grep -v grep | awk '{print $2}')
+  
+  build-finschia-docker-image:
+    runs-on: ubuntu-latest
+    needs: check-diffs
+    if: ${{ needs.check-diffs.outputs.get_diff }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/setup-buildx-action@v2
+      - name: Build and export
+        uses: docker/build-push-action@v4
+        with:
+          file: Dockerfile
+          context: .
+          tags: finschia/finschianode:latest
+          outputs: type=docker,dest=/tmp/finschia-ci-test.tar
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: finschia-ci-test
+          path: /tmp/finschia-ci-test.tar
+  
+  test-smart-contract-in-docker-container:
+    runs-on: ubuntu-latest
+    needs: [build-finschia-docker-image, get-cosmwasm-releases]
+    strategy:
+      fail-fast: false
+      matrix:
+        versions: ${{ fromJson(needs.get-cosmwasm-releases.outputs.cosmwasm_versions) }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/setup-buildx-action@v2
+      - name: Download smart contracts
+        uses: dsaltares/fetch-gh-release-asset@master
+        with:
+          repo: 'Finschia/cosmwasm'
+          version: tags/${{ matrix.versions }}
+          regex: true
+          file: '.*'
+          target: 'contracts/'
+      - name: Download docker image
+        uses: actions/download-artifact@v3
+        with:
+          name: finschia-ci-test
+          path: /tmp
+      - name: Load image
+        run: |
+          docker load --input /tmp/finschia-ci-test.tar
+          docker image ls -a
+      - name: Set configure file
+        run: ./init_single.sh docker
+        shell: bash
+      # sleep 5 seconds to wait for finschia to start up
+      - name: Start docker container
+        id: docker_start
+        run: |
+          container_id=$(docker run -d \
+            -p 26656:26656 -p 26657:26657 \
+            -v /home/runner/.finschia:/root/.finschia \
+            -v $(pwd)/ci_test:/root/ci_test \
+            -v $(pwd)/contracts:/root/contracts \
+            finschia/finschianode fnsad start)
```

Comment:
How about changing docker image name to special name? Because this image name is equals with pushed image name, we cannot recognize in this ci if the docker image is local built image or docker hub's image.

Reply:
That's what I'd like to do. However, init_single.sh currently uses [finschia/finschianode](https://github.com/170210/finschia/blob/feature/issue_226/init_single.sh#L12), and if I use another name to build docker image, it will cause init_single.sh to download the image from docker hub every time.

Reply:
How about using `fnsad` which is already built binary?

https://github.com/Finschia/finschia/pull/233/files#diff-2e2331a2bd1f068777d3ec06bbbcebb45c71c591fb0d6684f56d60174da639f5R49-R67